tools/fury.js

```diff
@@ -1,25 +1,73 @@
 var fs = require('fs');
 var path = require('path');
 var parser = require('../lib/fury').legacyBlueprintParser;
+var async = require('async');
+
+// TODO: Turn into a CLI argument
+var PRINT_PARSE_RESULT = false;
 
 // Process arguments
 var args = process.argv.slice(2);
 if (typeof args == 'undefined' || args.length !== 1) {
   var scriptName = path.basename(process.argv[1]);
-  console.log('usage: ' + scriptName + ' <input file>\n');
+  console.log('usage: ' + scriptName + ' <input file>|<input directory>\n');
   process.exit(0);
 }
 
-// Read & parse
-fs.readFile(args[0], 'utf8', function (err, data) {
-  if (err) throw err;
+// Creates parsing functions to be run in series
+function createParseFunction(filePath) {
+  return function(callback) {
+    var data = fs.readFileSync(filePath, 'utf8');
+    console.log('processing ' + filePath + ' ...');
+    parser.parse({ code: data }, function(error, api, warnings) {
+      if (error) {
 
-  parser.parse({ code: data }, function(error, api, warnings) {
-    if (error) {
-        console.log(JSON.stringify(error, null, 2));
-        process.exit(error.code);
-    }
+        return callback(error);
+      }
+
+      // For debug puproses
+      if (PRINT_PARSE_RESULT)
+        console.log(JSON.stringify(api, null, 2));
+      return callback();
+    });
+  };
+}
 
-    console.log(JSON.stringify(api, null, 2));
+// Walk through a directory and process every in it file
+function collectDirectory(dirPath) {
+  var parseFunctions = [];
+
+  fs.readdirSync(dirPath).forEach(function(name) {
+    var filePath = path.join(dirPath, name);
+    var stat = fs.statSync(filePath);
+    if (stat.isFile() && (path.extname(filePath) === '.apib'
+                          || path.extname(filePath) === '.md')) {
+      parseFunctions.push(createParseFunction(filePath));
+    }
+    // Do not recurse into directories for now
+    // else if (stat.isDirectory()) {
+    //     processDirectory(filePath, callback);
+    // }
   });
+
+  return parseFunctions;
+}
+
+// Query input
+var series = [];
+var stats = fs.statSync(args[0]);
+if (stats.isFile()) {
+  series.push(createParseFunction(args[0]));
+}
+else if (stats.isDirectory()) {
+  series = collectDirectory(args[0]);
+}
+
+console.log('processing ' + series.length + ' file(s)');
+async.series(series, function(error) {
+  if (error) {
+    console.log(error);
+  }
+
+  console.log('\ndone.');
```

Comment:
Should this be in an else block? So we don't print "done" when it errors? (maybe the intentions are it always says done).